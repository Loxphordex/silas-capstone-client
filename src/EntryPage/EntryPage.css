textarea {
  resize: none;
  min-width: 80%;
  height: 150px;
  border: 4px solid lightgray;
}

#quotes-area {
  height: 288px;
}

.quotebox {
  text-align: left;
}

.main-wrap label {
  border-radius: 50%;
  font-size: 24px;
  padding-bottom: 3px;
}

.sidebar {
    background-color: black;
    width: 150px;
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
}

.sidebar h1 {
    color: white;
}

.portfolio {
    background-color: rgb(32, 32, 32);
    padding: 5%;
    position: absolute;
    top: 0;
    left: 150px;
    right: 0;
    bottom: 0;
    
    -moz-transition: left 0.5s ease;
    transition: left 0.5s ease;
}

input[type=checkbox] {
   display: none;
}

input:checked ~ .portfolio { 
    left: 0;
}

input:checked ~ label {
    left: 0;
}

label {
    z-index: 2;
    position: absolute;
    top: 0;
    left: 150px;
    background-color: rgb(190, 190, 190);
    color: rgb(32, 32, 32);
    width: 36px;
    
    -moz-transition: left 0.5s ease;
    transition: left 0.5s ease;
}