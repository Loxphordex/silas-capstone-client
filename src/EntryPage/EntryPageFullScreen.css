@media only screen and (min-width: 600px) {
  nav {
    height: 60px;
  }
  #qnav h1 {
    padding-top: 10px;
  }
  .portfolio {
    background-color: white;
  }
  #portfolio-main {
    position: relative;
    top: 27px;
    margin: auto;
    width: 600px;
    min-width: 0%;
    box-shadow: 0 22px 22px 0 rgba(0, 0, 0, 0.2), 0 12px 20px 0 rgba(0, 0, 0, 0.19)
  }
  #quotes-area {
    width: 500px;
    min-height: 240px;
    margin: 0;
    margin-top: 0;
    display: inline-block;
  }
  #entry-area {
    width: 600px;
    margin: 0;
    display: inline-block;
  }
  #big-Q {
    top: -111px;
    left: 21px;
  }
  textarea {
    font-size: 16px;
  }
  .sidebar {
    min-height: 632px;
  }
  #logout {
    color: rgb(32, 32, 32);
    border-radius: 7px;
    font-size: 16px;
    min-width: 120px;
    position: absolute;
    top: 660px;
    right: 90px;
  }
}