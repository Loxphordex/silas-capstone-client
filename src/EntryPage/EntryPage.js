import React from 'react'
import './EntryPage.css'
import './EntryPageFullScreen.css'
import TokenService from '../services/token-service'
import ApiService from '../services/api-service'
import UserEntryList from '../UserEntryList/UserEntryList'
import Nav from '../Nav/Nav'

export default class EntryPage extends React.Component {

  handleLogout() {
    TokenService.clearAuthToken()
    this.props.resetState()
    this.props.clearError()
    this.props.history.history.push('/')
  }

  handleSaveEntry = (event) => {
    event.preventDefault()
    const content = event.target.entryText.value
    const title = event.target.title.value

    const entry = { title, content }
    console.log('TRIED TO GET ENTRIES')

    // Send the entry to the server
    ApiService.postEntry(entry)
      .then(() => {
        ApiService.getUserEntries()
        .then(entries => {
          this.props.updateUserEntries(entries)
        })
        .catch(err => this.props.handleError(err))        
      })
      .catch(err => this.props.handleError(err))
  }

  handleGetEntry = (entryId) => {
    console.log(entryId)
    ApiService.getEntryById(entryId)
      .then(entry => {
        this.props.updateEntry(entry.content)
        this.props.updateTitle(entry.title)
      })
  }

  render() {
    return (
      <div className='entry-page'>
        <div className="main-wrap">

            <input id="slide-sidebar" type="checkbox" checked='unchecked' readOnly={true} role="button" />
                <label htmlFor="slide-sidebar"><span>&#9776;</span></label>
            <div className="sidebar">
              <h2>My Journals</h2>
              <UserEntryList 
                userEntries={this.props.userEntries}
                updateUserEntries={this.props.updateUserEntries}
                updateEntry={this.props.updateEntry}
                updateTitle={this.props.updateTitle}
                handleGetEntry={this.handleGetEntry}
              />
              <button id='logout' onClick={() => this.handleLogout()}>Logout</button>
            </div>

            <div className="portfolio">
              <nav>
                <Nav />
              </nav>

              <div id='portfolio-main'>
              <section id='quotes-area'>
                <div id='big-Q'>Q</div>
                <div className='quotebox'>
                  <p className='quote-paragraph'>{
                    this.props.quotes && this.props.quotes.length > 0 &&
                    this.props.quotes[this.props.quotes.length - 1].quote
                  }</p>
                  <div className='quote-author'>{
                    this.props.quotes && this.props.quotes.length > 0 &&
                    '- ' + this.props.quotes[this.props.quotes.length - 1].author
                  }</div>
                </div>
              </section>

              <section id='entry-area'>
                <form id='entry_form' onSubmit={(event) => this.handleSaveEntry(event)}>
                  <input 
                    id='title' 
                    name='title' 
                    defaultValue={this.props.currentTitle}
                    placeholder='Title'></input>
                  <textarea 
                    placeholder='Type anything...'
                    id='entryText' 
                    name='entryText' 
                    value={this.props.currentEntry}
                    onChange={(event) => this.props.updateEntry(event.target.value)}>
                  </textarea>
                  <button type='submit' id='save-button'>Save</button>
                </form>  
              </section>            
              </div>
            </div>
        </div>        
      </div>
    )
  }
}