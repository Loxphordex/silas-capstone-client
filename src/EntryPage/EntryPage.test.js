import React from 'react';
import EntryPage from './EntryPage'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('Entry Page', () => {

  it('renders the complete page', () => {
    const wrapper = shallow(<EntryPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('initially renders no quotes', () => {
    const wrapper = mount(<EntryPage />)

    wrapper.setProps({ name: 'Moe' })

    expect(wrapper.props('name')).toEqual({ name: 'Moe' })
    expect(wrapper.containsMatchingElement(
      <button type='submit' id='save-button'>Save</button>
    )).toBeTruthy()
    expect(wrapper.find('#save-button').text()).toEqual('Save')
    expect(wrapper.find('.quote-paragraph').text()).toEqual('')
    expect(wrapper.find('.quote-author').text()).toEqual('')
  })
})