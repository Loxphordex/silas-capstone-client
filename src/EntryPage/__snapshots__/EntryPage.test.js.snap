// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry Page renders the complete page 1`] = `
<div
  className="entry-page"
>
  <div
    className="main-wrap"
  >
    <input
      defaultChecked="true"
      id="slide-sidebar"
      role="button"
      type="checkbox"
    />
    <label
      htmlFor="slide-sidebar"
    >
      <span>
        â˜°
      </span>
    </label>
    <div
      className="sidebar"
    >
      <h2>
        My Journals
      </h2>
      <UserEntryList
        handleGetEntry={[Function]}
      />
      <button
        id="logout"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
    <div
      className="portfolio"
    >
      <nav>
        <Nav />
      </nav>
      <div
        id="portfolio-main"
      >
        <section
          id="quotes-area"
        >
          <div
            id="big-Q"
          >
            Q
          </div>
          <div
            className="quotebox"
          >
            <p
              className="quote-paragraph"
            />
            <div
              className="quote-author"
            />
          </div>
        </section>
        <section
          id="entry-area"
        >
          <form
            id="entry_form"
            onSubmit={[Function]}
          >
            <input
              id="title"
              name="title"
              placeholder="Title"
            />
            <textarea
              id="entryText"
              name="entryText"
              onChange={[Function]}
              placeholder="Type anything..."
            />
            <button
              id="save-button"
              type="submit"
            >
              Save
            </button>
          </form>
        </section>
      </div>
    </div>
  </div>
</div>
`;
